testrepository/tests/test_testcommand.py

```diff
@@ -435,15 +604,90 @@ def test_run_tests_with_instances(self):
             ('popen', ('foo ',), {'shell': True, 'stdin': -1, 'stdout': -1})],
             ui.outputs)
 
+    def test_run_tests_with_profiles(self):
+        # when there are profiles and no instance_execute, run_tests runs
+        # profile cmds directly.
+        self.set_config(dedent("""\
+            [DEFAULT]
+            test_command=foo $IDLIST $PROFILE
+            list_profiles=probe
+            """))
+        ui, command = self.get_test_ui_and_cmd2()
+        ui.proc_outputs = [b'P1 P2', b'', b'']
+        self.useFixture(command)
+        fixture = self.useFixture(command.get_run_command(
+            test_ids={'1': {'profiles': ['P1']}, '2': {'profiles': ['P2']}}))
+        procs = fixture.run_tests()
+        self.expectThat(ui.outputs, MatchesAny(Equals([
+            ('popen', ('probe',), {'shell': True, 'stdin': -1, 'stdout': -1}),
+            ('communicate',),
+            ('values', [('running', _u('foo 2 P2'))]),
+            ('popen', (_u('foo 2 P2'),), {'shell': True, 'stdin': -1, 'stdout': -1}),
+            ('values', [('running', _u('foo 1 P1'))]),
+            ('popen', (_u('foo 1 P1'),), {'shell': True, 'stdin': -1, 'stdout': -1})
+            ]), Equals([
+            ('popen', ('probe',), {'shell': True, 'stdin': -1, 'stdout': -1}),
+            ('communicate',),
+            ('values', [('running', _u('foo 1 P1'))]),
+            ('popen', (_u('foo 1 P1'),), {'shell': True, 'stdin': -1, 'stdout': -1}),
+            ('values', [('running', _u('foo 2 P2'))]),
+            ('popen', (_u('foo 2 P2'),), {'shell': True, 'stdin': -1, 'stdout': -1}),
+            ])))
```

Comment:
Is there a reason why this uses `_u('...')` and elsewhere it uses the unicode literal `u'...'` ?